src/autofetcher.ts

```diff
@@ -327,5 +329,20 @@ export class AutoFetcher extends BackgroundBehavior {
 
     text.replace(STYLE_REGEX, urlExtractor).replace(IMPORT_REGEX, urlExtractor);
   }
+
+  extractDataAttributes(document) {
+    const allElements = document.querySelectorAll('*');
```

Comment:
CSS selectors don't allow querying by attribute name start, but Xpath does!

I think it might be more efficient to do this with xpath, for example, the following snippet can be pasted into the browser.

```
function* xpathNodes(path, root) {
  root = root || document;
  let iter = document.evaluate(path, root, null, XPathResult.ORDERED_NODE_ITERATOR_TYPE);
  let result = null;
  while ((result = iter.iterateNext()) !== null) {
    yield result;
  }
}

for (const res of xpathNodes("//@*[starts-with(name(), 'data-') and (starts-with(., 'http') or starts-with(., '/') or starts-with(., './') or starts-with(., '../'))]")) {
    console.log(res.value);
}
```

The xpathNodes is already defined and can be imported from `utils`

Reply:
Ended up implementing this to try it out, pushed to the PR!